connectors/byoc.py

```diff
@@ -578,39 +511,42 @@ def next_sync(self):
         if self.sync_now:
             logger.debug("sync_now is true, syncing!")
             return 0
-        if not self.scheduling["enabled"]:
+        if not self.scheduling.get("enabled", False):
             logger.debug("scheduler is disabled")
             return SYNC_DISABLED
-        return next_run(self.scheduling["interval"])
+        return next_run(self.scheduling.get("interval"))
+
+    async def reset_sync_now_flag(self):
+        await self.index.update(doc_id=self.id, doc={"sync_now": False})
+
+    async def sync_starts(self):
+        doc = {
+            "last_sync_status": JobStatus.IN_PROGRESS.value,
+            "last_sync_error": None,
+            "status": Status.CONNECTED.value,
+        }
+        await self.index.update(doc_id=self.id, doc=doc)
 
     async def _sync_starts(self, sync_job_index):
         job_id = await sync_job_index.create(self)
+        if self.sync_now:
+            await self.reset_sync_now_flag()
+
         sync_job = await sync_job_index.fetch_by_id(job_id)
         await sync_job.claim()
+        await self.sync_starts()
 
-        self.sync_now = self.doc_source["sync_now"] = False
-        self.doc_source["last_sync_status"] = JobStatus.IN_PROGRESS.value
-        self.status = Status.CONNECTED
-        await self.sync_doc()
-
-        self._start_time = time.time()
         logger.info(f"Sync starts, Job id: {job_id}")
         return await sync_job_index.fetch_by_id(job_id)
 
     async def error(self, error):
-        self.doc_source["error"] = str(error)
-        await self.sync_doc()
-
-    async def _sync_suspended(self, job):
-        await job.suspend()
-        self.doc_source["last_sync_status"] = job.status.value
-        self.doc_source["last_sync_error"] = None
-        self.doc_source["error"] = None
-        self.doc_source["last_synced"] = iso_utc()
-        await self.sync_doc()
-        logger.info(f"Sync suspended, Job id: {job.job_id}")
-
-    async def _sync_done(self, job, status, result, exception=None):
```

Comment:
`_sync_done` will eventually be removed once job execution is moved out of `Connector` class in a separate PR.

Reply:
Why is `_sync_suspended` deleted?

Reply:
`_sync_suspended` is replaced by calling
https://github.com/elastic/connectors-python/blob/544cfea7eeb6bd874592155afd73405a6d917b8d/connectors/byoc.py#L781